.github/workflows/pull_request.yml

```diff
@@ -3,6 +3,10 @@ name: PR
 on:
   pull_request
 
+permissions:
+  contents: read
+  pull-requests: read
```

Comment:
How did you figure out you didn't need the read and write permissions for pull-requests? Did you just try the least privilege and it worked? Is there some documentation about it?

Reply:
I used the least privilege first. Some sections in our workflows had permissions already added prior to these PR's so I based of that. 
If a GA does not have the correct permissions the workflow should fail :
https://graphite.dev/guides/github-actions-permissions#:~:text=Creating%20directories%20within%20GitHub%20Actions%20might%20fail%20due%20to%20insufficient%20write%20permissions%20in%20the%20working%20directory.%20To%20fix%20this%2C%20adjust%20your%20workflow%20to%20modify%20permissions%20or%20choose%20a%20directory%20where%20the%20runner%20has%20write%20access%3A